gnd/src/main/java/com/google/android/gnd/ui/offlinearea/viewer/OfflineAreaViewerFragment.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.ui.offlinearea.viewer;
+
+import static com.google.android.gnd.rx.RxAutoDispose.autoDisposable;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import butterknife.OnClick;
+import com.google.android.gnd.MainActivity;
+import com.google.android.gnd.R;
+import com.google.android.gnd.databinding.OfflineAreaViewerFragBinding;
+import com.google.android.gnd.model.basemap.OfflineArea;
+import com.google.android.gnd.model.feature.Point;
+import com.google.android.gnd.ui.common.AbstractFragment;
+import com.google.android.gnd.ui.map.MapAdapter;
+import com.google.android.gnd.ui.map.MapProvider;
+import io.reactivex.Single;
+import javax.inject.Inject;
+
+public class OfflineAreaViewerFragment extends AbstractFragment {
+
+  private static final String MAP_FRAGMENT = MapProvider.class.getName() + "#fragment";
+
+  @Inject MapProvider mapProvider;
+
+  private OfflineAreaViewerViewModel viewModel;
+  @Nullable private MapAdapter map;
+
+  public static OfflineAreaViewerFragment newInstance() {
```

Comment:
Can we use Dagger to instantiate this instead? Injecting `Provider<OfflineAreaViewerFragment>` into the class where it's needed should work, maybe...

Reply:
Done, it was already `@Binds` annotated so all we needed was to add `@Inject`! bd5a85e